### PR TITLE
Feature group testing

### PR DESCRIPTION
**Pull request by BriannaRoskind**
Summary:
Generated Feature Group controller code and wrote passing integration tests for the Feature Group controller.

Files Added:
- test/e2e/tests/test_feature_group.py - Contains Create and Delete integration test for feature group
- test/e2e/resources/feature_group.yaml - yaml file for feature group creation in integration testing.
- pkg/resource/feature_group/custom_delta.go - Sets default spec values for feature group fields (DisableGlueTableCreation and S3StorageConfig) if the user did not define them.

Files Edited:
- test/e2e/service_bootstrap.py - Attached AmazonSageMakerFeatureStoreAccess Role Policy to the SageMaker Execution Role for testing purposes. (Feature Store requires this permission for the Feature Group creation and access)
- generator.yaml - Commented out feature group from the ignore list. Also added feature group exceptions, hooks, and fields.
- pkg/resource/feature_group/delta.go - added the CustumSetDefault() function from custom_delta.go to the newResourceDelta() function.

### CODE REVIEW

File: test/e2e/service_bootstrap.py
Comment (surajkota) [inline test/e2e/service_bootstrap.py:61]: did you check what is present in this policy that is not in SagemakerFullAccess?

File: pkg/resource/feature_group/custom_delta.go
Comment (surajkota) [inline pkg/resource/feature_group/custom_delta.go:42]: little confusing to see a variable called resolved is part of a create operation. what is the use case for this field to be input by user?

File: test/e2e/tests/test_feature_group.py
Comment (surajkota) [inline test/e2e/tests/test_feature_group.py:119]: FYI: you are not completely blocked on this
Have you looked at this sample PR? - https://github.com/aws-controllers-k8s/elasticache-controller/pull/34/files

You can implement the logic to requeueOnDelete. only this missing is the resource getting patched after delete is called. This will be enabled once [the PR on runtime is merged](https://github.com/aws-controllers-k8s/runtime/pull/21/files)

File: test/e2e/tests/test_feature_group.py
Comment (surajkota) [inline test/e2e/tests/test_feature_group.py:112]: TODO: add resource side checks

**Review by surajkota — CHANGES_REQUESTED** [review]
few comments

File: test/e2e/service_bootstrap.py
Comment (BriannaRoskind) [reply test/e2e/service_bootstrap.py:61]: I examined the two policies and included in the [ACK Feature Group](https://quip-amazon.com/MNHYAu4tEaRT/ACK-Feature-Group-Updated#QMc9CAmH6aG) quip doc what the SageMakerFeatureStoreAccess has that SageMakerFullAccess does not.

File: generator.yaml
Comment (mbaijal) [inline generator.yaml:513]: did we decide to change this column name across resources ?

File: test/e2e/tests/test_feature_group.py
Comment (mbaijal) [inline test/e2e/tests/test_feature_group.py:33]: nit: confusing variable name

File: test/e2e/tests/test_feature_group.py
Comment (mbaijal) [reply test/e2e/tests/test_feature_group.py:112]: why is this a TODO? Is it not planned for this PR ?

File: generator.yaml
Comment (BriannaRoskind) [reply generator.yaml:513]: This update happened in [PR #44](https://github.com/aws-controllers-k8s/sagemaker-controller/pull/44/files).

File: test/e2e/tests/test_feature_group.py
Comment (BriannaRoskind) [reply test/e2e/tests/test_feature_group.py:121]: Added a TODO for implementing the logic to requeueOnDelete.

File: test/e2e/tests/test_feature_group.py
Comment (BriannaRoskind) [reply test/e2e/tests/test_feature_group.py]: I have now replaced RESOURCE_NAME_BASE with RESOURCE_NAME_PREFIX.

File: apis/v1alpha1/ack-generate-metadata.yaml
Comment (A-Hilaly) [inline apis/v1alpha1/ack-generate-metadata.yaml:8]: To keep this version up-to-date you'll have to regenerate the controller using `AWS_SDK_GO_VERSION=v1.38.11` environment variable :)

File: test/e2e/tests/test_feature_group.py
Comment (surajkota) [reply test/e2e/tests/test_feature_group.py:112]: yes new PR

**Review by surajkota — COMMENTED** [review]
@BriannaRoskind one comment related to iam role
please squash your commits(you have 22) because ack bot doesnt do it :)
and create an issue for the fields getting added to unrelated shapes so we don't forget

apart from that lgtm, get a second approval

File: test/e2e/service_bootstrap.py
Comment (surajkota) [reply test/e2e/service_bootstrap.py:61]: looked at it and I think you shouldn't need it because [sagemaker execution role has S3 full access](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/test/e2e/service_bootstrap.py#L55-L57)

please remove it and see

**Review by surajkota — APPROVED** [review]
/approve

**Comment by surajkota** [comment]
Please squash your commits so we can merge once tests pass

**Comment by surajkota** [comment]
add clean up here as well - https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/test/canary/scripts/run_test.sh#L30

**Pull request merged**
